File: scripts/list_models.py
import os
import google.generativeai as genai

api_key = os.getenv("GEMINI_API_KEY")
genai.configure(api_key=api_key)

print("Listing supported models for your API key...")
for m in genai.list_models():
    if 'generateContent' in m.supported_generation_methods or 'image' in m.name:
        print(f"Model: {m.name}, Methods: {m.supported_generation_methods}")

File: scripts/list_models.mjs
import { GoogleGenAI } from '@google/genai';
const ai = new GoogleGenAI({});
async function main() {
    const models = await ai.models.list();
    for await (const model of models) {
        if (model.name.includes("image") || model.name.includes("imagen")) {
            console.log(model.name);
        }
    }
}
main();
